src/App.tsx

```diff
@@ -2,26 +2,69 @@ import React from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export class App extends React.Component {
+  state = {
+    goods: [],
+  };
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  render(): React.ReactNode {
+    const { goods } = this.state;
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    return (
+      <div className="App">
+        <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+        <button
+          type="button"
+          data-cy="all-button"
+          onClick={() => {
```

Comment:
Consider extracting the onClick function to a separate method for better readability and maintainability.